### PR TITLE
[actions] rpi changes

### PR DESCRIPTION
**Pull request by dimitre**
This PR intends to fix RPI github actions
[hopefully] modernizing and fixing cross compiling issues
it replaces compilers installed via ci.openframeworks.cc with apt installable ones (and cached via github actions)

Status:
- [x] aarch64 emptyexample
- [x] aarch64 alladdonsexample
- [x] armv6l emptyexample
- [x] armv6l alladdonsexample

Errors - all fixed:
~~Using armv6l actions it doesnt find <bcm_host.h>, it is not installable via firmware.zip anymore.
Maybe it is possible to install via apt packages, have to research more about this one.

aarch64 alladdonsexample linking errors:
```
/usr/lib/gcc-cross/aarch64-linux-gnu/11/../../../../aarch64-linux-gnu/bin/ld: warning: liblapack.so.3, needed by /home/runner/work/openFrameworks/openFrameworks/scripts/ci/linuxrpi/raspbian/usr/lib/aarch64-linux-gnu/libopencv_core.so, not found (try using -rpath or -rpath-link)
/usr/lib/gcc-cross/aarch64-linux-gnu/11/../../../../aarch64-linux-gnu/bin/ld: warning: libblas.so.3, needed by /home/runner/work/openFrameworks/openFrameworks/scripts/ci/linuxrpi/raspbian/usr/lib/aarch64-linux-gnu/libopencv_core.so, not found (try using -rpath or -rpath-link)
/home/runner/work/openFrameworks/openFrameworks/scripts/ci/linuxrpi/raspbian/usr/lib/aarch64-linux-gnu/libm.a(s_sin.o): in function `__sin':
(.text+0x10): relocation truncated to fit: R_AARCH64_LD64_GOTPAGE_LO15 against symbol `__stack_chk_guard@@GLIBC_2.17' defined in .data.rel.ro section in /home/runner/work/openFrameworks/openFrameworks/scripts/ci/linuxrpi/raspbian/lib/ld-linux-aarch64.so.1
/usr/lib/gcc-cross/aarch64-linux-gnu/11/../../../../aarch64-linux-gnu/bin/ld: (.text+0x10): warning: too many GOT entries for -fpic, please recompile with -fPIC
collect2: error: ld returned 1 exit status
```
~~

### CODE REVIEW

**Comment by artificiel** [comment]
not in a Pi position right now but for `bcm_host.h` if these changes have implications that go "way back", a few years ago it became necessary to install `libraspberrypi-dev` and maybe `raspberrypi-kernel-headers` to get things in general to compile.

**Comment by dimitre** [comment]
Thanks. for what I've seen there are some headers in this repo / folder
https://github.com/raspberrypi/userland
host_applications/linux/libs/bcm_host/include

**Comment by ofTheo** [comment]
@dimitre - I'll have a look! 
I think I tried doing things without firmware.zip and ran into the same issues with `bcm_host.h`

**Comment by artificiel** [comment]
for liblaplack (CV dependency), is `liblapack-dev` already in there?

depending on how deep you want to dig maybe it's simpler to keep the old pre-packaged toolchain for armv6 and your updated approach from aarch64?

**Comment by dimitre** [comment]
I suggest we drop armv6l architecture since aarch64 will run in most RPI versions including
Pi Zero 2
Pi 3
Pi 4
Pi 5

**Comment by artificiel** [comment]
+1 for dropping armv6l, but it brings back the idea of having a table of "which version of OF supports which old(er) platforms". it has to be simple to identify and download the binary dist of common legacy systems.

(and/or keeping the old actions in place for armv6 until they really stop working? — as I understand, this issue is an attempt at modernizing the toolchain, presumably to tackle the performance issues with the graphics in Pi4/5, but did the previous setup work as expected for armv6? I remember some fiddling was required around that custom "downloaded" toolchain to get C++17 support on said armv6 actions; if we made it there, it may keep working for a while? and yes move to up-to-date for aarch64).

**Comment by dimitre** [comment]
now armv6l uses EGL approach which is considered ancient and deprecated. 
The good news is Linux 6.8 kernel now has video drivers for RPI 5, so things will be easier and more performant.

armv6l is almost working, it is only a fact of downloading and linking correctly bcm_host.h. it is using default toolchains installable with apt-get which turns things easier, faster (github cdn) and less costs for openframeworks.cc server.

**Comment by dimitre** [comment]
@danoli3 can help us making opencv work with or without lib Lapack and Blas.
I've seen in apothecary one parameter like
```      -DWITH_LAPACK=OFF \ ```
but this is for other platforms, 
in Linux we use apt installed ones, I suppose they are built with Lapack and Blas by default
the other option is install them and link them correctly

**Comment by ofTheo** [comment]
@dimitre what if you add this line to the addon_config.mk for `linuxaarch64`

```

linuxaarch64:
	ADDON_PKG_CONFIG_LIBRARIES = opencv4
	ADDON_LIBS_EXCLUDE = libs/opencv/%
	ADDON_INCLUDES_EXCLUDE = libs/opencv
	ADDON_INCLUDES_EXCLUDE += libs/opencv/%
	ADDON_LDFLAGS =  -lblas -llapack  <--- add this line 

```

**Comment by dimitre** [comment]
Still failing, but I think if we can pass two paths also it can hopefully complete. 
This two libraries have individual folders
like:
```
/usr/lib/arch64-linux-gnu/blas
/usr/lib/arch64-linux-gnu/lapack
```
and 
```
/usr/lib/arm-linux-gnueabihf/blas
/usr/lib/arm-linux-gnueabihf/lapack
```
@ofTheo any ideas of what lines to add in addon_config.mk ?

**Comment by dimitre** [comment]
bcm_host.h solved and other fixes. it is finishing compilation here locally, 
still issues on allAddons, only a matter of passing the correct library paths I think

File: libs/openFrameworksCompiled/project/linuxarmv6l/config.linuxarmv6l.default.mk [hunk 1]
```diff
@@ -205,10 +208,8 @@ endif
 ################################################################################
 
 # Broadcom hardware interface library
-PLATFORM_HEADER_SEARCH_PATHS += $(RPI_ROOT)/opt/vc/include
-PLATFORM_HEADER_SEARCH_PATHS += $(RPI_ROOT)/opt/vc/include/IL
-PLATFORM_HEADER_SEARCH_PATHS += $(RPI_ROOT)/opt/vc/include/interface/vcos/pthreads
-PLATFORM_HEADER_SEARCH_PATHS += $(RPI_ROOT)/opt/vc/include/interface/vmcs_host/linux
+PLATFORM_HEADER_SEARCH_PATHS += $(RPI_ROOT)/userland/host_applications/linux/libs/bcm_host/include
+PLATFORM_HEADER_SEARCH_PATHS += $(RPI_ROOT)/userland
```
Comment (ofTheo) [inline libs/openFrameworksCompiled/project/linuxarmv6l/config.linuxarmv6l.default.mk:212]: Will this work for people working with the device itself ( as opposed to building in the CI )? 
ie: does userland exist for Pi device users?

**Comment by ofTheo** [comment]
congrats @dimitre !! 
ooof - that was a gnarly one.

File: libs/openFrameworksCompiled/project/linuxarmv6l/config.linuxarmv6l.default.mk (hunk 1 above)
Comment (dimitre) [reply libs/openFrameworksCompiled/project/linuxarmv6l/config.linuxarmv6l.default.mk:212]: Good catch. maybe I should put back the includes with /opt/vc in armv6l
easy to do

Comment (dimitre) [reply libs/openFrameworksCompiled/project/linuxarmv6l/config.linuxarmv6l.default.mk:212]: and enclose this in ifdef CROSS_COMPILE

**Pull request merged**